model/generate.go

```diff
@@ -270,6 +256,29 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, cachedProj
 	if err := AddNewBuilds(ctx, true, v, p, newTVPairsForNewVariants, g.TaskID); err != nil {
 		return errors.Wrap(err, "errors adding new builds")
 	}
+
+	newTasks := append(newTVPairsForExistingVariants.ExecTasks, newTVPairsForNewVariants.ExecTasks...)
+	if err = addDependencies(t, p, v, newTasks); err != nil {
+		return errors.Wrap(err, "error adding dependencies")
+	}
+
+	return nil
+}
+
+func addDependencies(t *task.Task, p *Project, v *Version, newTasks TVPairSet) error {
+	taskIDTable := NewTaskIdTable(p, v, "", "")
```

Comment:
Instead of using this to see what the task IDs were, does it make more sense to have (as an example) AddNewTasks return the build that it added tasks to, and get the IDs from there?

Reply:
done.
I decided to not get the ids from the build because I think the task cache doesn't include exec tasks. Instead, I returned a slice of task ids.